src/Npgsql/Internal/TypeHandlers/ByteaHandler.cs

```diff
@@ -43,19 +44,44 @@ public override async ValueTask<byte[]> Read(NpgsqlReadBuffer buf, int len, bool
 
     ValueTask<ArraySegment<byte>> INpgsqlTypeHandler<ArraySegment<byte>>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription)
         => throw new NotSupportedException("Only writing ArraySegment<byte> to PostgreSQL bytea is supported, no reading.");
+    
+    ValueTask<Stream> INpgsqlTypeHandler<Stream>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription)
+        => throw new NotSupportedException("Only writing Stream to PostgreSQL bytea is supported, no reading.");
```

Comment:
Too bad we can't easily call NpgsqlDataReader.GetStream from here!

```suggestion
        => throw new NotSupportedException("Reading a PostgreSQL bytea as a Stream is unsupported, use NpgsqlDataReader.GetStream() instead..");
```